lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -757,6 +808,77 @@ Duration calculateDuration(
     return result;
   }
 
+  private void addErrorsToResultObject(
+      String owner,
+      BulkOperationResults<String, TaskanaException> bulkLog,
+      List<MinimalTaskSummary> existingMinimalTaskSummaries) {
+    for (MinimalTaskSummary taskSummary : existingMinimalTaskSummaries) {
+      if (!owner.equals(taskSummary.getOwner())) { // owner was not set
+        if (!taskSummary.getTaskState().equals(TaskState.READY)) { // due to invalid state
+          bulkLog.addError(
+              taskSummary.getTaskId(),
+              new InvalidStateException(
+                  "Task "
+                      + taskSummary.getTaskId()
+                      + " is in state "
+                      + taskSummary.getTaskState()));
+        } else { // due to unknown reason
+          bulkLog.addError(
+              taskSummary.getTaskId(),
+              new UpdateFailedException(
+                  "Could not set owner of Task " + taskSummary.getTaskId() + "."));
+        }
+      }
+    }
+  }
+
+  private void handleNonExistingTasks(
```

Comment:
Personally I think we could be more precise in the naming. "AddExceptionForNonExistingTasks" for example. And isn't it a good practice to return the modified value? In this case the bulklog?

Reply:
Done. I changed all new methods so that they return the bulklog that they create locally.